### ISSUE TITLE
Download dialogue from new tabs fails with "close unused blank tabs" enabled

### ISSUE THREAD

**Issue by jhard** [char101/tabgroupsbtn#6]
When you have a site that links to a file to download and uses target=_blank for doing so, a new tab will open and once the response begins, it closes and the "what to do with ..." dialogue shows up.

It doesn't work with Tab Groups Button, instead I get errors in the console 

`TypeError: aBrowser.adjustPriority is not a function NetworkPrioritizer.jsm:88:4

Handler function DebuggerProgressListener.prototype.onStateChange threw an exception: [Exception... "Component returned failure code: 0x80004002 (NS_NOINTERFACE) [nsIWebProgress.DOMWindow]"  nsresult: "0x80004002 (NS_NOINTERFACE)"  location: "JS frame :: resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/devtools/server/actors/webbrowser.js :: DebuggerProgressListener.prototype.onStateChange< :: line 2130"  data: no]
Stack: DebuggerProgressListener.prototype.onStateChange<@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/devtools/server/actors/webbrowser.js:2130:9
makeInfallible/<@resource://gre/modules/commonjs/toolkit/loader.js -> resource://gre/modules/devtools/DevToolsUtils.js:82:14

Line: 2130, column: 0 DevToolsUtils.js:58:0
Missing window information when showing nsIHelperAppLauncherDialog: [Exception... "Component returned failure code: 0x80004002 (NS_NOINTERFACE) [nsIInterfaceRequestor.getInterface]"  nsresult: "0x80004002 (NS_NOINTERFACE)"  location: "JS frame :: resource://gre/components/nsHelperAppDlg.js :: nsUnknownContentTypeDialog.prototype.show :: line 146"  data: no] nsHelperAppDlg.js`

If I disable the "close unsed blank tabs" option, it works like it should.

Took quite a while to figure out TGB was the culprit, but now that I know a workaround, this is low prio for me. <3 the addon.

**Comment by char101** [comment]
I never experience that problem.

Why do you need the `target=_blank` attribute, if the attachment type is `download`, the download dialog should appear without changing the current URL.

**Comment by jhard** [comment]
I don't need the attribute, but sites use it ;)
I'll try to whip up a test-case later today or tomorrow & look at using a fresh profile to make sure it's not some combination of reasons.

**Comment by char101** [comment]
I see. Since the error comes from the developer tools, I assumed that you were testing some application. 

Anyway, I found these bug report

https://bugzilla.mozilla.org/show_bug.cgi?id=1162693

I suspect the error comes from firefox itself. This addon only happens to do the tab closing.

Also there are addons which provide the function to close blank tab when downloading a file, so this addon and the other addon could try to close the tab in parallel.

**Comment by jhard** [comment]
Yes, I found that bug, too. I agree that the error is thrown by firefox itself, TGB is just the trigger.

I'll look into plugin combos & report back.
